src/main/java/io/vertx/core/logging/LoggerFactory.java

```diff
@@ -46,18 +46,55 @@ public static synchronized void initialise() {
     }
 
     if (className != null) {
-      ClassLoader loader = Thread.currentThread().getContextClassLoader();
+      ClassLoader tcccl = Thread.currentThread().getContextClassLoader();
       try {
-        Class<?> clz = loader.loadClass(className);
-        delegateFactory = (LogDelegateFactory) clz.newInstance();
-      } catch (Exception e) {
-        throw new IllegalArgumentException("Error instantiating transformer class \"" + className + "\"", e);
+        newDelegateFactory = loadClass(className, tcccl);
+      } catch (IllegalArgumentException e) {
+        ClassLoader thisClasssLoader = LoggerFactory.class.getClassLoader();
+        newDelegateFactory = loadClass(className, thisClasssLoader);
       }
     } else {
-      delegateFactory = new JULLogDelegateFactory();
+      newDelegateFactory = new JULLogDelegateFactory();
     }
 
-    LoggerFactory.delegateFactory = delegateFactory;
+    LoggerFactory.delegateFactory = newDelegateFactory;
+  }
+
+  private static LogDelegateFactory loadClass(String className, ClassLoader classLoader) {
+    try {
+      Class<?> clz = classLoader.loadClass(className);
+      return (LogDelegateFactory) clz.newInstance();
+    } catch (Exception e) {
+      throw new IllegalArgumentException("Error instantiating transformer class \"" + className + "\"", e);
+    }
+  }
+
+  /**
+   * Set new custom LoggerFactory.
+   * @param factory the new LogDelegateFactory to use
+   * @author Michael Vorburger.ch
+   */
+  public static void setLogDelegateFactory(LogDelegateFactory factory) {
+      LogDelegateFactory current = delegateFactory;
+      if (current != null) {
+          try {
+            getLogger(LoggerFactory.class).info("Replacing LogDelegateFactory " + current + " by " + factory);
+          } catch (Throwable t) {
+            // Too bad...
+            t.printStackTrace();
```

Comment:
Can't we use a logger of `factory` to manage this stack trace?

Reply:
duh, yeah ... nice idea - I simply didn't think of it; now done as suggested